src/MaxText/layers/attention_op.py

```diff
@@ -1330,9 +1462,27 @@ def apply_attention_dot(
     # Casting softmaxt computation for float32 for model stability.
     if self.float32_logits:
       attn_weights = attn_weights.astype(jnp.float32)
+
     attn_mask = self.generate_attention_mask(
         query, key, decoder_segment_ids, model_mode, previous_chunk, bidirectional_mask
     )
+    if hasattr(self.config, "moba") and self.config.moba:
```

Comment:
Why do we need to check `hasattr(self.config, "moba")`? This config is added to base.yml